qdrant-landing/content/documentation/concepts/hybrid-queries.md

```diff
@@ -0,0 +1,1066 @@
+        limit=100,
+    ),
+    query=[
+        [0.1, 0.2, ...],  # <─┐
+        [0.2, 0.1, ...],  # < ├─ multi-vector
+        [0.8, 0.9, ...],  # < ┘
+    ],
+    using="colbert",
+    limit=10,
+)
+```
+
+```typescript
+import { QdrantClient } from "@qdrant/js-client-rest";
+
+const client = new QdrantClient({ host: "localhost", port: 6333 });
+
+client.query("{collection_name}", {
+    prefetch: {
+        query: [1, 23, 45, 67], // <------------- small byte vector
+        limit: 100,
+    },
+    query: [
+        [0.1, 0.2], // <─┐
+        [0.2, 0.1], // < ├─ multi-vector
+        [0.8, 0.9], // < ┘
+    ],
+    using: 'colbert',
+    limit: 10,
+});
+```
+
+```rust
+use qdrant_client::Qdrant;
+use qdrant_client::qdrant::{PrefetchQueryBuilder, Query, QueryPointsBuilder};
+
+let client = Qdrant::from_url("http://localhost:6334").build()?;
+
+client.query(
+    QueryPointsBuilder::new("{collection_name}")
+        .add_prefetch(PrefetchQueryBuilder::default()
+            .query(Query::new_nearest(vec![0.01, 0.45, 0.67]))
+            .limit(100u64)
+        )
+        .query(Query::new_nearest(vec![
+            vec![0.1, 0.2],
+            vec![0.2, 0.1],
+            vec![0.8, 0.9],
+        ]))
+        .using("colbert")
+        .limit(10u64)
+).await?;
+```
+
+```java
+import static io.qdrant.client.QueryFactory.nearest;
+
+import io.qdrant.client.QdrantClient;
+import io.qdrant.client.QdrantGrpcClient;
+import io.qdrant.client.grpc.Points.PrefetchQuery;
+import io.qdrant.client.grpc.Points.QueryPoints;
+
+
+QdrantClient client =
+    new QdrantClient(QdrantGrpcClient.newBuilder("localhost", 6334, false).build());
+
+client
+    .queryAsync(
+        QueryPoints.newBuilder()
+            .setCollectionName("{collection_name}")
+            .addPrefetch(
+                PrefetchQuery.newBuilder()
+                    .setQuery(nearest(0.01f, 0.45f, 0.67f)) // <-- dense vector
+                    .setLimit(100)
+                    .build())
+            .setQuery(
+                nearest(
+                    new float[][] {
+                      {0.1f, 0.2f},	// <─┐
+                      {0.2f, 0.1f},	// < ├─ multi-vector
+                      {0.8f, 0.9f}	// < ┘
+                    }))
+            .setUsing("colbert")
+            .setLimit(10)
+            .build())
+    .get();
+```
+
+```csharp
+using Qdrant.Client;
+using Qdrant.Client.Grpc;
+
+var client = new QdrantClient("localhost", 6334);
+
+await client.QueryAsync(
+  collectionName: "{collection_name}",
+  prefetch: new List <PrefetchQuery> {
+    new() {
+      Query = new float[] { 0.01f, 0.45f, 0.67f	},	// <-- dense vector****
+        Limit = 100
+    }
+  },
+  query: new float[][] {
+    [0.1f, 0.2f], // <─┐
+    [0.2f, 0.1f], // < ├─ multi-vector
+    [0.8f, 0.9f]  // < ┘
+  },
+  usingVector: "colbert",
+  limit: 10
+);
+```
+
+Even more sophisticated examples like leveraging all the above techniques in a single query are possible:
```

Comment:
Is it the first time we mention that prefetch can be nested?

Reply:
yes

Reply:
I think it would good to have somewhere an explicit explanation of the execution model regarding:
- prefetches can be nested
- nested prefetches are executed bottom up
- prefetches at the same level have their results concatenated

WDYT?

Reply:
Ok, let me check where to put this